server/app/views/admin/programs/ProgramPredicateConfigureView.java

```diff
@@ -523,9 +523,9 @@ private DivTag createScalarDropdown(
             .withText("Sorry, you cannot create a show/hide predicate with this question type.");
       }
     }
-
     ImmutableList<OptionTag> options =
         scalars.stream()
+            .filter(scalar -> !scalar.equals(Scalar.NAME_SUFFIX))
```

Comment:
**suggestion:** Instead of doing this here, could we add a condition above for the name question where we do the `if (questionDefinition.isAddress())` - that way all the special cases are in one place? So it would be something like `if (questionDefinition.isAddress())` `else if (questionDefinition.isName()`..

Reply:
sure! added a condition above for the name question to filter out name_suffix before hand.